api/types.go

```diff
@@ -157,6 +157,32 @@ func (tr Timerange) IsValid() bool {
 		tr.Start <= tr.End
 }
 
+func rounder(x, r int64) int64 {
+	if x < 0 {
+		return -rounder(-x, r)
+	}
+	// This performs a round
+	return (x + r/2) / r * r
```

Comment:
Some extremely large numbers aren't exactly representable as `float64` (2^53 or about around 10^16) so I'm a little concerned about introducing floating point error. In practice I wouldn't expect it to be a concern, though.